src/storage/core.rs

```diff
@@ -872,11 +872,9 @@ impl Inner {
         if safe.active_blob.is_none() {
             Err(Error::active_blob_doesnt_exist().into())
         } else {
-            // FIXME: write lock is still held, so everyone will wait for dump, maybe it's better
-            // to derive this operation to `try_dump_old_blob_indexes`
-            safe.active_blob.as_mut().unwrap(/*None case checked*/).dump().await?;
```

Comment:
Agreed, the logic of dumping indexes should be similar. But since we don't have observer at this level, I've added observer call to the indexes dumping to the upper level. Now indexes dumping is a separate observer command